exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/NetworkFunctions.java

```diff
@@ -0,0 +1,619 @@
+    nulls = FunctionTemplate.NullHandling.NULL_IF_NULL
+  )
+  public static class InetNtoaFunction implements DrillSimpleFunc {
+
+    @Param
+    BigIntHolder in;
+
+    @Output
+    VarCharHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+
+    public void setup() {
+    }
+
+
+    public void eval() {
+      StringBuilder result = new StringBuilder(15);
+
+      long inputInt = in.value;
+
+      for (int i = 0; i < 4; i++) {
+
+        result.insert(0,Long.toString(inputInt & 0xff));
+
+        if (i < 3) {
+          result.insert(0,'.');
+        }
+
+        inputInt = inputInt >> 8;
+      }
+
+      String outputValue = result.toString();
+
+      out.buffer = buffer;
+      out.start = 0;
+      out.end = outputValue.getBytes().length;
+      buffer.setBytes(0, outputValue.getBytes());
+    }
+
+
+  }
+
+  /**
+   * This function returns true if a given IPv4 address is private, false if not.
+   */
+
+  @FunctionTemplate(
+    name = "is_private_ip",
+    scope = FunctionTemplate.FunctionScope.SIMPLE,
+    nulls = FunctionTemplate.NullHandling.NULL_IF_NULL
+  )
+  public static class IsPrivateIP implements DrillSimpleFunc {
+
+    @Param
+    VarCharHolder inputTextA;
+
+    @Output BitHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+
+    public void setup() {
+    }
+
+
+    public void eval() {
+      String ipString = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(inputTextA.start, inputTextA.end, inputTextA.buffer);
+
+      String[] ipAddressInArray = ipString.split("\\.");
+
+      int result = 0;
+
+      int[] octets = new int[3];
+
+      for( int i = 0; i < 3; i++ ){
+        octets[i] = Integer.parseInt( ipAddressInArray[i] );
+        if( octets[i] > 255 || octets[i] < 0 ) {
+          result = 0;
+        }
+      }
+
+      if( octets[0] == 192 && octets[1] == 168 ) {
+        result = 1;
+      }
+      else if (octets[0] == 172 && octets [1] >= 16 && octets[1] <= 31 ){
+        result = 1;
+      }
+      else if( octets[0] == 10 ) {
+        result = 1;
+      }
+      else {
+        result = 0;
+      }
+
+      out.value = result;
+    }
+  }
+  /**
+   *  This function converts an IPv4 address into a BigInt.  Useful for sorting IPs, or looking for IP ranges.
+   *  IE:
+   *  SELECT *
+   *  FROM <data>
+   *  ORDER BY inet_aton( ip ) ASC
+   */
+  @FunctionTemplate(
+    name = "inet_aton",
+    scope = FunctionTemplate.FunctionScope.SIMPLE,
+    nulls = FunctionTemplate.NullHandling.NULL_IF_NULL
+  )
+  public static class InetAtonFunction implements DrillSimpleFunc {
+
+    @Param
+    VarCharHolder inputTextA;
+
+    @Output BigIntHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+
+    public void setup() {
+    }
+
+
+    public void eval() {
+      String ipString = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(inputTextA.start, inputTextA.end, inputTextA.buffer);
+      if( ipString == null || ipString.isEmpty() || ipString.length() == 0 ){
```

Comment:
`isEmpty()` is the same as `length() == 0`. Please remove excess check.
Please fix here and in other functions